csi/controller.go

```diff
@@ -378,7 +379,11 @@ func (s *OsdCsiServer) CreateVolume(
 	var newVolumeId string
 	if source.Parent == "" {
 		// Get Capabilities and Size
-		spec.Shared = csiRequestsSharedVolume(req)
+		if spec.Shared && !spec.Sharedv4 {
+			spec.Shared = csiRequestsSharedVolume(req)
+		} else {
+			spec.Sharedv4 = csiRequestsSharedVolume(req)
+		}
```

Comment:
Added legacy support for `shared` volumes if `shared=true` is set.  @adityadani 

`shared` volume are only used if 
`shared=true` and `sharedv4=false (or sharedv4 is empty)`  

All other cases result in sharedv4:
* default: no SC params but ReadWriteMany mode set
* or, when sharedv4=true

Reply:
perfect!